platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/EventStreamMultiFileIterator.java

```diff
@@ -54,46 +56,38 @@ public class EventStreamMultiFileIterator implements IOIterator<DetailedConsensu
+     * Create an iterator that walks over events in an event stream spanning multiple files, starting from the indicated
+     * event stream bound.  The bound must occur in the event stream otherwise a NoSuchElementException will be thrown.
      *
-     * @param fileIterator
-     * 		an iterator that returns ordered event stream files
-     * @param startingRound
-     * 		do not return events prior to this round, or {@link EventStreamPathIterator#FIRST_ROUND_AVAILABLE}
-     * 		if all events should be walked by this iterator
-     * @throws IOException
-     * 		if there is a problem reading the event stream
-     * @throws NoSuchElementException
-     * 		if the starting round can't be found
+     * @param fileIterator an iterator that returns ordered event stream files
+     * @param bound        the lower bound of the events in the event stream to be walked by this iterator.
+     * @throws IOException            if there is a problem reading the event stream
+     * @throws NoSuchElementException if an event prior to or matching the bound can't be found
      */
-    public EventStreamMultiFileIterator(final Iterator<Path> fileIterator, final long startingRound)
-            throws IOException {
-
-        this.fileIterator = fileIterator;
+    public EventStreamMultiFileIterator(
+            @NonNull final Iterator<Path> fileIterator, @NonNull final EventStreamBound bound) throws IOException {
+        Objects.requireNonNull(bound, "the lower bound must not be null");
+        this.fileIterator = Objects.requireNonNull(fileIterator, "the file iterator must not be null");
         this.startHash = null;
         this.skippedEvents = new ArrayList<>();
 
-        // Remove events from before the requested round
-        while (hasNext() && peek().getConsensusData().getRoundReceived() < startingRound) {
+        // Remove events from before the requested bound
+        while (hasNext() && bound.compareTo(peek(), BoundType.LOWER) < 0) {
             skippedEvents.add(next());
         }
     }
 
     /**
-     * Create an iterator that walks over all events in a given directory starting with a given round.
+     * Create an iterator that walks over all events in a given directory starting from a given event stream bound. The bound must occur in the event stream otherwise a NoSuchElementException will be thrown.
      *
-     * @param eventStreamDirectory
-     * 		the directory in question
-     * @param startingRound
-     * 		the round to start with, or {@link EventStreamPathIterator#FIRST_ROUND_AVAILABLE}
-     * 		if all events should be walked by this iterator
-     * @throws IOException
-     * 		if there is a problem reading the event stream
-     * @throws NoSuchElementException
-     * 		if the starting round can't be found
+     * @param eventStreamDirectory the directory in question
+     * @param bound                the  bound for the events in the event stream to be walked by this iterator
```

Comment:
nit: extra space between `the` and `bound`

Reply:
removed